schema/crdb/dbinit.sql

```diff
@@ -2513,20 +2514,40 @@ CREATE TABLE omicron.public.switch_port_settings_address_config (
 /*******************************************************************/
 
 /*
- * Metadata for the schema itself.  This version number isn't great, as there's
+ * Metadata for the schema itself. This version number isn't great, as there's
  * nothing to ensure it gets bumped when it should be, but it's a start.
  */
 
-CREATE TABLE omicron.public.db_metadata (
-    name  STRING(63) NOT NULL PRIMARY KEY,
-    value STRING(1023) NOT NULL
+CREATE TABLE IF NOT EXISTS omicron.public.db_metadata (
+    -- There should only be one row of this table for the whole DB.
+    -- It's a little goofy, but filter on "singleton = true" before querying
+    -- or applying updates, and you'll access the singleton row.
+    --
+    -- We also add a constraint on this table to ensure it's not possible to
+    -- access the version of this table with "singleton = false".
```

Comment:
I didn't follow that, but I may once I've looped back through everything.

Reply:
This one is straightforward -- we just want a table with one row.

- If we didn't have the `singleton` column: Someone could insert several rows into this DB, which would be confusing (which one would be valid?)
- If we didn't have the `singleton` constraint, we'd be allowed **two** rows (one for "true", one for "false"). This limits it to one.